### PR TITLE
vfloat sequencing

### PR DESCRIPTION
**Pull request by orion160**

### CODE REVIEW

**Comment by orion160** [comment]
Prior to merge needs UopGen refactor

File: arches/isa_json/gen_uarch_rv64v_json.py [hunk 1]
```diff
@@ -217,24 +217,165 @@
...
+# Vector Floating-Point Instructions: Vector Single-Width Floating-Point Add/Subtract Instructions
+    "vfadd.vv" :  {"pipe" : "vfloat", "uop_gen" : "ARITH", "latency" : 6},
+    "vfadd.vf" :  {"pipe" : "vfloat", "uop_gen" : "ARITH", "latency" : 6},
+    
+    "vfsub.vv" :  {"pipe" : "vfloat", "uop_gen" : "ARITH", "latency" : 6},
+    "vfsub.vf" :  {"pipe" : "vfloat", "uop_gen" : "ARITH", "latency" : 6},
+    "vfrsub.vf" :  {"pipe" : "vfloat", "uop_gen" : "ARITH", "latency" : 6},
+
+# Vector Floating-Point Instructions: Vector Widening Floating-Point Add/Subtract Instructions
+    "vfwadd.vv" :  {"pipe" : "vfloat", "uop_gen" : "ARITH", "latency" : 6},
```
Comment (kathlenemagnus-mips) [inline arches/isa_json/gen_uarch_rv64v_json.py:229]: These instructions have wide dests so they should use `ARITH_WIDE_DEST`.

File: arches/isa_json/gen_uarch_rv64v_json.py [hunk 2]
```diff
@@ -217,24 +217,165 @@
...
+    "vfnmadd.vv" :  {"pipe" : "vfloat", "uop_gen" : "ARITH", "latency" : 6},
+
+    "vfmsub.vv" :  {"pipe" : "vfloat", "uop_gen" : "ARITH", "latency" : 6},
+    "vfmsub.vf" :  {"pipe" : "vfloat", "uop_gen" : "ARITH", "latency" : 6},
+
+    "vfnmsub.vv" :  {"pipe" : "vfloat", "uop_gen" : "ARITH", "latency" : 6},
+    "vfnmsub.vf" :  {"pipe" : "vfloat", "uop_gen" : "ARITH", "latency" : 6},
+
+# Vector Floating-Point Instructions: Vector Widening Floating-Point Fused Multiply-Add Instructions
+    "vfwmacc.vv" :  {"pipe" : "vfloat", "uop_gen" : "ARITH", "latency" : 6},
```
Comment (kathlenemagnus-mips) [inline arches/isa_json/gen_uarch_rv64v_json.py:280]: These should use the `ARITH_MAC` which adds the destination of each uop as a source.

**Comment by kathlenemagnus-mips** [comment]
@orion160 the uop generators for the vector floating point instructions need to be reviewed. I would also add a new execution unit for `vfmul` to be analogous to the `vmul` unit.

File: arches/isa_json/gen_uarch_rv64v_json.py [hunk 3]
```diff
@@ -194,19 +194,19 @@
     "vmv.v.i" : {"pipe" : "vint", "uop_gen" : "ARITH", "latency" : 1},
 
 # Vector Fixed-Point Arithmetic Instructions: Vector Single-Width Saturating Add and Subtract
-    "vsaddu.vv" : {"pipe" : "vfixed", "uop_gen" : "ARITH", "latency" : 1},
-    "vsaddu.vx" : {"pipe" : "vfixed", "uop_gen" : "ARITH", "latency" : 1},
-    "vsaddu.vi" : {"pipe" : "vfixed", "uop_gen" : "ARITH", "latency" : 1},
+    "vsaddu.vv" : {"pipe" : "vint", "uop_gen" : "ARITH", "latency" : 1},
```
[the remark this answers is not in the document]
Comment (orion160) [reply arches/isa_json/gen_uarch_rv64v_json.py:197]: In the merged commit there was vfixed but only in the json.

I'll be doing a branch from which all the sequencing part is done.
And to dont be blocked until the reformat is merged

**Comment by orion160** [comment]
@kathlenemagnus-mips should it be the same with vfdiv?

I think that the float division, and the reciprocal estimates could make use of this resource.

**Pull request closed without merging**